stream/client.go

```diff
@@ -93,20 +96,26 @@ func (c *Client) StreamSearch(ctx context.Context, q query.Q, opts *zoekt.Search
 			}
 		case eventError:
 			if errString, ok := reply.Data.(string); ok {
-				if strings.Contains(errString, "context canceled") {
-					return context.Canceled
-				} else if strings.Contains(errString, "context deadline exceeded") {
-					return context.DeadlineExceeded
-				} else {
-					return fmt.Errorf(errString)
+				if ctxErr, ok := stringToContextError(errString); ok {
```

Comment:
surprising to me that the client cancelling the context leads to us getting a context error string from the server. I would suspect that instead we should have some sort of well formed error for timeouts. If it is us cancelling then either the server shouldn't return an error, or the client should be getting the error. Not sure if I'm making sense, so I am sure we will chat about this over a call :)

Reply:
The part you are commenting on is related to `gob` not supporting `error`. As a workaround, we convert context errors to strings on server-sdie and here we convert them back to proper context errors.

Reply:
EDIT: The behavior of the stream client here is similar to that of `http.Client.Do`: `Do` also returns a context error if the context was cancelled on client-side.